# Settings for ros_control control loop
hardware_control_loop:
   loop_hz: &loop_hz 125

# Settings for ros_control hardware interface
ur_hardware_interface:
   joints: &robot_joints
     - shoulder_pan_joint
     - shoulder_lift_joint
     - elbow_joint
     - wrist_1_joint
     - wrist_2_joint
     - wrist_3_joint

# Publish all joint states ----------------------------------
joint_state_controller:
   type: joint_state_controller/JointStateController
   publish_rate: *loop_hz

arm_controller:
   type: velocity_controllers/JointTrajectoryController
   joints: *robot_joints
   stop_trajectory_duration: 0.5
   state_publish_rate: *loop_hz
   action_monitor_rate: 20
   constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      shoulder_pan_joint: {trajectory: 0.2, goal: 0.1}
      shoulder_lift_joint: {trajectory: 0.2, goal: 0.1}
      elbow_joint: {trajectory: 0.2, goal: 0.1}
      wrist_1_joint: {trajectory: 0.2, goal: 0.1}
      wrist_2_joint: {trajectory: 0.2, goal: 0.1}
      wrist_3_joint: {trajectory: 0.2, goal: 0.1}
   gains: &gain
      shoulder_pan_joint:
         p: 10
         d: 1
         i: 1
         i_clamp: 1
      shoulder_lift_joint:
         p: 10
         d: 1
         i: 1
         i_clamp: 1
      elbow_joint:
         p: 10
         d: 1
         i: 1
         i_clamp: 1
      wrist_1_joint:
         p: 10
         d: 1
         i: 1
         i_clamp: 1
      wrist_2_joint:
         p: 10
         d: 1
         i: 1
         i_clamp: 1
      wrist_3_joint:
         p: 10
         d: 1
         i: 1
         i_clamp: 1

joint_group_vel_controller:
   type: velocity_controllers/JointGroupVelocityController
   joints: *robot_joints
   gains: *gain

joint_group_pos_controller:
   type: position_controllers/JointGroupPositionController
   joints: *robot_joints
   gains: *gain